path-walk.c

```diff
@@ -280,6 +280,28 @@ int walk_objects_by_path(struct path_walk_info *info)
 	if (prepare_revision_walk(info->revs))
 		die(_("failed to setup revision walk"));
 
+	if (info->edge_aggressive) {
+		for (struct commit_list *list = info->revs->commits;
```

Comment:
I am not familiar enough with the `info->revs` structure: do the `commits` include _all_ of the walked commits, or only the ones specified by the original `argv`? If it is the latter, we might miss boundary (shallow) commits that we _would_ want to handle.

Reply:
These are all the commits at the tip of the walk. I will more carefully mention this in the future commit messages and PR description, but this is trying to mimic the behavior of the [existing references to `edge_hint_aggressive`](https://github.com/microsoft/git/blob/d347ab22a8377f7780b47e0ea93161300415c9ec/list-objects.c#L307-L335): these walk the "tip" commits that are marked UNINTERESTING before the commit walk begins (even if they don't show up in the boundary after the walk).